lib/query/builder.js

```diff
@@ -1241,7 +1250,11 @@ assign(Builder.prototype, {
 
   // Helper function for clearing or reseting a grouping type from the builder
   _clearGrouping(grouping) {
-    this._statements = reject(this._statements, { grouping });
+    if(grouping in this._single) {
+      delete this._single[grouping];
```

Comment:
Is there a particular reason to use `delete` over "set to undefined" here? Delete is slow.